### DISCUSSION TITLE
CORS policy error

### DISCUSSION THREAD

**Discussion by ghost** [firebase/firebase-js-sdk#5592] [Q&A] [answered]
Hello,

I apologize if this question is very good (I am very new to JavaScript and Firebase). I have been trying to set up Firebase by following this tutorial: [Getting started with Firebase for the web -- Firebase Fundamentals](https://www.youtube.com/watch?v=rQvOAnNvcNQ). So far, it has been helpful, but when I run index.html, I get the following error: 

![Inkedi1_LI](https://user-images.githubusercontent.com/88161092/136583365-46741a08-de40-4bd5-a6dd-e143ba8996dc.jpg)

I am not sure how to solve this error, but it seems like the CORS policy is blocking my access to the firebase libraries.

In case this helps, here is my index.js code:
```
import { initializeApp } from "https://www.gstatic.com/firebasejs/8.10.0/firebase-app.js";
import { getAuth, onAuthStateChanged } from "https://www.gstatic.com/firebasejs/8.10.0/firebase-auth.js";
import { getFirestore } from 'https://www.gstatic.com/firebasejs/8.10.0/firebase-firestore.js';

// Your web app's Firebase configuration

// For Firebase JS SDK v7.20.0 and later, measurementId is optional

const firebaseApp = initializeApp({

    apiKey: "apiKey information",

    authDomain: "authDomain information",

    databaseURL: "databaseURL information",

    projectId: "projectId information",

    storageBucket: "storageBucket information",

    messagingSenderId: "messagingSenderId information",

    appId: "appId information",

    measurementId: "measurementId information"

});

const auth = getAuth(firebaseApp);
const db = getFirestore(firebaseApp);

// Detect auth state
onAuthStateChanged(auth, user => {
    if (user != null) {
        console.log('logged in!');
    }
    else {
        console.log('No user');
    }
})
```

And my index.html code:

```<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <title>Title</title>
    <script type="module" src="index.js"> </script>
</head>
<body>

</body>
</html>
```

**Comment by hsubox76** [comment, marked as answer]
The "x-ijt" header that is interfering with the request seems to be something Webstorm adds: https://youtrack.jetbrains.com/issue/WEB-30191 and not related to Firebase. I'd suggest asking in Webstorm forums or Stack Overflow, or anywhere else you can find help from Webstorm users.

**Comment by ghost** [comment]
@hsubox76 Hi Christina, thank you for your suggestion and for pointing me in the right direction! 

I actually think that I may have found the solution to my problem and nearly burst into tears out of relief. I was looking up if someone received a similar error using Webstorm, and I found a post that helped me solve this error: [CORS errors from Font Awesome font in Angular app](https://stackoverflow.com/questions/57822107/cors-errors-from-font-awesome-font-in-angular-app). 

It seems that going into settings and clicking "Allow unsigned requests" will help. Thank you again, and I hope you have a wonderful week!

Michelle
